run/connect_cloud_sql/main.tf

```diff
@@ -141,61 +141,53 @@ resource "google_secret_manager_secret_iam_member" "secretaccess_compute_dbname"
-        # Sets a environment variable for instance connection name
-        env {
-          name  = "INSTANCE_CONNECTION_NAME"
-          value = google_sql_database_instance.mysql_instance.connection_name
-        }
-        # Sets a secret environment variable for database user secret
-        env {
-          name = "DB_USER"
-          value_from {
-            secret_key_ref {
-              name = google_secret_manager_secret.dbuser.secret_id # secret name
-              key  = "latest"                                      # secret version number or 'latest'
-            }
+    containers {
+      image = "us-docker.pkg.dev/cloudrun/container/hello:latest" # Image to deploy
+      # Sets a environment variable for instance connection name
+      env {
+        name  = "INSTANCE_CONNECTION_NAME"
+        value = google_sql_database_instance.mysql_instance.connection_name
+      }
+      # Sets a secret environment variable for database user secret
+      env {
+        name = "DB_USER"
+        value_source {
+          secret_key_ref {
+            secret  = google_secret_manager_secret.dbuser.secret_id # secret name
+            version = "latest"                                      # secret version number or 'latest'
           }
         }
-        # Sets a secret environment variable for database password secret
-        env {
-          name = "DB_PASS"
-          value_from {
-            secret_key_ref {
-              name = google_secret_manager_secret.dbpass.secret_id # secret name
-              key  = "latest"                                      # secret version number or 'latest'
-            }
+      }
+      # Sets a secret environment variable for database password secret
+      env {
+        name = "DB_PASS"
+        value_source {
+          secret_key_ref {
+            secret  = google_secret_manager_secret.dbpass.secret_id # secret name
+            version = "latest"                                      # secret version number or 'latest'
           }
         }
-        # Sets a secret environment variable for database name secret
-        env {
-          name = "DB_NAME"
-          value_from {
-            secret_key_ref {
-              name = google_secret_manager_secret.dbname.secret_id # secret name
-              key  = "latest"                                      # secret version number or 'latest'
-            }
+      }
+      # Sets a secret environment variable for database name secret
+      env {
+        name = "DB_NAME"
+        value_source {
+          secret_key_ref {
+            secret  = google_secret_manager_secret.dbname.secret_id # secret name
+            version = "latest"                                      # secret version number or 'latest'
           }
         }
       }
     }
-
-    metadata {
-      annotations = {
-        "run.googleapis.com/client-name" = "terraform"
-      }
-    }
   }
 
-  autogenerate_revision_name = true
```

Comment:
Query: As per below docs, `autogenerate_revision_name` does not seem to be same as `client`. Won't the users be missing out this feature?

* https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloud_run_service
* https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloud_run_v2_service

Reply:
This wasn't intended as a replacement. It just so happens the metadata annotations were above this block, so it looks like it's a replacement in the diff. 

A lack of defining a `revision` implies autogeneration in v2